### PR TITLE
#5372: Add end to end demo for functional stable diffusion model

### PR DESCRIPTION
**Pull request by kkeerthana0573**
This PR has end to end demo for the functional stable diffusion model.

### CODE REVIEW

File: models/experimental/functional_stable_diffusion/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## functional_stable_diffusion Demo
+## How to Run
+
```
Comment (tt-rkim) [inline models/experimental/functional_stable_diffusion/README.md:3]: Please put a clarification stating that the user must build the project, activate the environment, and set the appropriate environment variables.

File: tt_metal/python_env/requirements-dev.txt [hunk 2]
```diff
@@ -23,6 +23,8 @@ datasets==2.9.0
 torch==2.2.1.0+cpu
 networkx==3.1
 torchvision==0.17.1+cpu
+torchmetrics==1.3.1
```
Comment (tt-rkim) [inline tt_metal/python_env/requirements-dev.txt:26]: Please run fast dispatch post-commit, build all C++ binaries, and eager package on main and assert their correctness.

**Review by tt-rkim — CHANGES_REQUESTED** [review]

File: tt_metal/python_env/requirements-dev.txt (hunk 2 above)
Comment (kkeerthana0573) [reply tt_metal/python_env/requirements-dev.txt:26]: The mentioned CIs passed and here are the corresponding links -

- [ all - Fast dispatch post-commit main build ](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8078596332)
- [Build C++ binaries with all configs](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8078604735)
- [eager - Package on main](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8078612614)

File: models/experimental/functional_stable_diffusion/README.md (hunk 1 above)
Comment (kkeerthana0573) [reply models/experimental/functional_stable_diffusion/README.md:3]: Updated it. 
Thank you

**Comment by kkeerthana0573** [comment]
The demo is experiencing hang at various stages during each trigger on WH.

To be specific, The demo includes "run_demo_inference" and "run_demo_diffusion_db," both executed for 256x256 and 512x512 image sizes with 2 num_inference_steps.

During the initial run, the test was unresponsive for the 512x512 image size during the 2nd denoising step in "run_demo_diffusion_db."
In the subsequent run, it encountered a hang for the same image size during the 1st denoising step in "run_demo_diffusion_db."

Note: This demo hang issue is not limited to only run_demo_diffusion_db test, it is observed in run_demo_inference test at times.

**Review by arakhmati — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Comment by kkeerthana0573** [comment]
CIs passed. Link [here](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8155716782)

**Pull request merged**